src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeed.cs

```diff
@@ -46,23 +47,51 @@ public BlobFeed(string accountName, string accountKey, string containerName, str
             string url = $"{FeedContainerUrl}/{blobPath}?comp=metadata";
             using (HttpClient client = new HttpClient())
             {
-                client.DefaultRequestHeaders.Clear();
-                var request = AzureHelper.RequestMessage("GET", url, AccountName, AccountKey).Invoke();
-                using (HttpResponseMessage response = await client.SendAsync(request))
+                const int MaxAttempts = 15;
+                // add a bit of randomness to the retry delay
+                var rng = new Random();
+                int retryCount = MaxAttempts;
+
+                while (true)
                 {
-                    if (response.IsSuccessStatusCode)
+                    try
                     {
-                        Log.LogMessage(
-                            MessageImportance.Low,
-                            $"Blob {blobPath} exists for {AccountName}: Status Code:{response.StatusCode} Status Desc: {await response.Content.ReadAsStringAsync()}");
+                        client.DefaultRequestHeaders.Clear();
+                        var request = AzureHelper.RequestMessage("GET", url, AccountName, AccountKey).Invoke();
+                        using (HttpResponseMessage response = await client.SendAsync(request))
+                        {
+                            if (response.IsSuccessStatusCode)
+                            {
+                                Log.LogMessage(
+                                    MessageImportance.Low,
+                                    $"Blob {blobPath} exists for {AccountName}: Status Code:{response.StatusCode} Status Desc: {await response.Content.ReadAsStringAsync()}");
+                            }
+                            else
+                            {
+                                Log.LogMessage(
+                                    MessageImportance.Low,
+                                    $"Blob {blobPath} does not exist for {AccountName}: Status Code:{response.StatusCode} Status Desc: {await response.Content.ReadAsStringAsync()}");
+                            }
+                            return response.IsSuccessStatusCode;
```

Comment:
this isn't something you're adding here but this is a very strange way to decide something exists or not.